src/cider/nrepl/middleware/info.clj

```diff
@@ -308,12 +308,12 @@
 
 (defn eldoc-reply
   [msg]
-  (let [info (info msg)]
-    (if-let [arglists (extract-arglists info)]
-      (merge (extract-ns-or-class info)
-             (extract-name-or-member info)
-             {:eldoc (format-arglists arglists)})
-      {:status :no-eldoc})))
+  (if-let [info (info msg)]
+    (merge (extract-ns-or-class info)
+           (extract-name-or-member info)
+           {:doc (:doc info)}
+           {:eldoc (-> info extract-arglists format-arglists)})
+    {:status :no-eldoc}))
```

Comment:
Yep, got it :-) So the docstring will always be returned, even for functions. We could certainly use it somewhere in the eldoc. I'll add the `:type` key here.